passenger_car_dynamic_model/src/passenger_car_dynamic_model/PassengerCarDynamicModel.cpp

```diff
@@ -0,0 +1,283 @@
+  post_step_func_ = std::bind(&PassengerCarDynamicModel::ODEPostStep, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3, std::placeholders::_4, std::placeholders::_5);
+};
+
+PassengerCarDynamicModel::~PassengerCarDynamicModel() {};
+
+void PassengerCarDynamicModel::setParameterServer(std::shared_ptr<ParameterServer> parameter_server) {
+  param_server_ = parameter_server;
+  
+  // Load Parameters
+  bool l_f_param        = param_server_->getParam("length_to_f", l_f_);
+  bool l_r_param        = param_server_->getParam("length_to_r", l_r_);
+  bool R_f_param        = param_server_->getParam("effective_wheel_radius_f", R_ef_);
+  bool R_r_param        = param_server_->getParam("effective_wheel_radius_r", R_er_);
+  bool long_stiff_param = param_server_->getParam("tire_longitudinal_stiffness", C_sx_);
+  bool lat_stiff_param  = param_server_->getParam("tire_cornering_stiffness", C_ay_);
+  bool inertia_param    = param_server_->getParam("moment_of_inertia", I_z_);
+  bool mass_param       = param_server_->getParam("vehicle_mass", m_);
+
+  // Check if all the required parameters could be loaded
+  if (!(l_f_param && l_r_param && R_f_param && R_r_param
+         && long_stiff_param && lat_stiff_param && inertia_param && mass_param)) {
+
+    std::ostringstream msg;
+    msg << "One of the required parameters could not be found or read " 
+      << " length_to_f: " << l_f_param 
+      << " length_to_r: " << l_r_param 
+      << " effective_wheel_radius_f: " << R_f_param 
+      << " effective_wheel_radius_r: " << R_r_param 
+      << " tire_longitudinal_stiffness: " << long_stiff_param
+      << " tire_cornering_stiffness: " << lat_stiff_param 
+      << " moment_of_inertia: " << inertia_param
+      << " vehicle_mass: " << mass_param;
+
+    throw std::invalid_argument(msg.str());
+  }
+}
+
+std::vector<VehicleState> PassengerCarDynamicModel::predict(const VehicleState& initial_state,
+  double timestep, double delta_t) {
+
+    // Populate control inputs
+    // This predict function takes in no new control inputs so extract the old ones from the state vector
+    VehicleControlInput control_input;
+    control_input.target_steering_angle = initial_state.prev_steering_cmd;
+    control_input.target_velocity = initial_state.prev_vel_cmd;
+
+    // Ensure we run at least 1 step
+    size_t num_steps;
+    if (delta_t <= timestep) {
+      num_steps = 1;
+    } else {
+      num_steps = delta_t / timestep;
+    }
+
+    std::vector<VehicleControlInput> control_inputs(num_steps, control_input);
+
+    // Call default predict method
+    return predict(initial_state, control_inputs, timestep);
+  }
+
+std::vector<VehicleState> PassengerCarDynamicModel::predict(const VehicleState& initial_state,
+  const std::vector<VehicleControlInput>& control_inputs, double timestep) {
+        
+    // Construct output vector
+    std::vector<VehicleState> resulting_states;
+    resulting_states.reserve(control_inputs.size());
```

Comment:
You probably want to check the size of ```control_inputs``` before you proceed.

Reply:
I think an empty control_inputs list here is invalid input so I will add this check to the ConstraintChecker class